File: final_asssignment_backend_go/project/internal/domain/user_role.go
package domain

import (
	"gorm.io/gorm"
)

// UserRole 表示 user_role 表的实体
type UserRole struct {
	UserID    int            `gorm:"column:user_id;primaryKey" json:"user_id"`
	RoleID    int            `gorm:"column:role_id;primaryKey" json:"role_id"`
	DeletedAt gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (UserRole) TableName() string {
	return "user_role"
}

File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/UserRole.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("user_role")
public class UserRole implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    @TableField("user_id")
    private Integer userId;

    @TableField("role_id")
    private Integer roleId;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}